autoswitch_virtualenv.plugin.zsh

```diff
@@ -17,6 +17,22 @@ function maybeworkon() {
   fi
 }
 
+# Gives the path to the nearest parent .venv file or nothing if it gets to root
+function check_venv_path()
+{
+    local check_dir=$1
+
+    if [[ -f "${check_dir}/.venv" ]]; then
+        echo "${check_dir}/.venv"
+        return
+    else
+        if [ "$check_dir" = "/" ]; then
+            return
+        fi
+        check_venv_path $(dirname "$check_dir")
```

Comment:
```
## New branch
$ time (PWD=~/a/b/c/d/e/f/g/h/i/j/k/l/m/n/o/p/q/r/s/t/u/v/w/x/y/z MYOLDPWD=~/a/b/c check_venv)    
( PWD=~/a/b/c/d/e/f/g/h/i/j/k/l/m/n/o/p/q/r/s/t/u/v/w/x/y/z MYOLDPWD=~/a/b/c )  0.02s user 0.05s system 105% cpu 0.065 total

$ time (PWD=~/a MYOLDPWD=~/a/b/c check_venv)                                                      
( PWD=~/a MYOLDPWD=~/a/b/c check_venv; )  0.00s user 0.00s system 98% cpu 0.005 total             

## Old branch
$ time (PWD=~/a/b/c/d/e/f/g/h/i/j/k/l/m/n/o/p/q/r/s/t/u/v/w/x/y/z MYOLDPWD=~/a/b/c check_venv)    
( PWD=~/a/b/c/d/e/f/g/h/i/j/k/l/m/n/o/p/q/r/s/t/u/v/w/x/y/z MYOLDPWD=~/a/b/c )  0.00s user 0.00s system 64% cpu 0.001 total
```
So, it does significantly increase time for long directories. However, it's only 1/20th of a second and for short directories, it approaches the performance of the old version.